assignments/closure.js

```diff
@@ -4,19 +4,51 @@
 // that manipulates variables defined in the outer scope.
 // The outer scope can be a parent function, or the top level of the script.
 
+function food (diet){
+  let rice = 'carbohydrate';
+  let beans = 'protein';
+  
+  function nutrition(foodlover){
+      console.log(`${diet} ${rice} is consumed, you get energy`);
+  }
+return nutrition();
+};
+food('anytime');
+
+
 
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
 
 // ==== Challenge 2: Implement a "counter maker" function ====
 const counterMaker = () => {
+  let count = 0;
+  return function counter(){
+    count++;
+    if(count > 5){ //is it ojoro putting number 3 solution here? lol
+      return 1
+    }
+    return count;
+    
+  }
```

Comment:
This is good. Only thing is the question says 5 should be a parameter so that when you call counter maker, you can pass it any limit you want. for example:

```
const counterMaker = (limit) => {
  let count = 0
  const counter = () => {
    count++
    if (count > limit) {
      count = 1
    }
    return count
  }
  return counter
}
```